pkg/apiserver-impl/api_default_service.go

```diff
@@ -2,62 +2,78 @@ package apiserver_impl
 
 import (
 	"context"
-	"errors"
-	openapi "github.com/redhat-developer/odo/pkg/apiserver-gen/go"
+	"fmt"
 	"net/http"
+
+	openapi "github.com/redhat-developer/odo/pkg/apiserver-gen/go"
+	"github.com/redhat-developer/odo/pkg/component/describe"
+	"github.com/redhat-developer/odo/pkg/kclient"
+	odocontext "github.com/redhat-developer/odo/pkg/odo/context"
+	"github.com/redhat-developer/odo/pkg/podman"
+	"github.com/redhat-developer/odo/pkg/state"
 )
 
 // DefaultApiService is a service that implements the logic for the DefaultApiServicer
 // This service should implement the business logic for every endpoint for the DefaultApi API.
 // Include any external packages or services that will be required by this service.
 type DefaultApiService struct {
+	cancel       context.CancelFunc
+	pushWatcher  chan<- struct{}
+	kubeClient   kclient.ClientInterface
+	podmanClient podman.Client
+	stateClient  state.Client
 }
 
 // NewDefaultApiService creates a default api service
-func NewDefaultApiService() openapi.DefaultApiServicer {
-	return &DefaultApiService{}
+func NewDefaultApiService(
+	cancel context.CancelFunc,
+	pushWatcher chan<- struct{},
+	kubeClient kclient.ClientInterface,
+	podmanClient podman.Client,
+	stateClient state.Client,
+) openapi.DefaultApiServicer {
+	return &DefaultApiService{
+		cancel:       cancel,
+		pushWatcher:  pushWatcher,
+		kubeClient:   kubeClient,
+		podmanClient: podmanClient,
+		stateClient:  stateClient,
+	}
 }
 
 // ComponentCommandPost -
 func (s *DefaultApiService) ComponentCommandPost(ctx context.Context, componentCommandPostRequest openapi.ComponentCommandPostRequest) (openapi.ImplResponse, error) {
-	// TODO - update ComponentCommandPost with the required logic for this service method.
-	// Add api_default_service.go to the .openapi-generator-ignore to avoid overwriting this service implementation when updating open api generation.
-
-	// TODO: Uncomment the next line to return response Response(200, GeneralSuccess{}) or use other options such as http.Ok ...
-	// return Response(200, GeneralSuccess{}), nil
-
-	return openapi.Response(http.StatusNotImplemented, nil), errors.New("ComponentCommandPost method not implemented")
+	switch componentCommandPostRequest.Name {
+	case "push":
+		s.pushWatcher <- struct{}{}
+		return openapi.Response(http.StatusOK, nil), nil
```

Comment:
I think the response body should contain a message like when the operation has been submitted successfully. At least, that's what I can see from the response example provided in the example:
```json
{
  "message": "push was successfully executed"
}
```